### PR TITLE
Update on repair

### PR DESCRIPTION
**Pull request by ilonalit**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/13622

Add support for forcing update on repair in instance lifecycle policy in compute_instance_group_manager and compute_region_instance_group_manager resources (Beta).

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added field `instance_lifecycle_policy` to `google_compute_instance_group_manager` and `google_compute_region_instance_group_manager` (beta)
```
```release-note:enhancement
compute: added field `force_update_on_repair` to `instance_lifecycle_policy` (beta)
```

### CODE REVIEW

**Comment by ilonalit** [comment]
Looks like the test **TestAccRegionInstanceGroupManager_stateful** has been already failing. It is not related to this feature. AFAIK it should be fixed in another pull request soon, I would wait.

**Comment by rileykarson** [comment]
Adding @hao-nan-li as an initial reviewer. They'll make a first pass, then hand off final review to @c2thorn.

File: mmv1/third_party/terraform/tests/resource_compute_instance_group_manager_test.go.erb [hunk 1]
```diff
@@ -630,6 +630,12 @@ resource "google_compute_instance_group_manager" "igm-update" {
     }
   }
 <% end -%>
+
+<% unless version == "ga" -%>
```
Comment (hao-nan-li) [inline mmv1/third_party/terraform/tests/resource_compute_instance_group_manager_test.go.erb:634]: Is this field updatable? If yes can we make sure to test that?

**Comment by hao-nan-li** [comment]
> Looks like the test **TestAccRegionInstanceGroupManager_stateful** has been already failing. It is not related to this feature. AFAIK it should be fixed in another pull request soon, I would wait.

This failing test is flaky, not related to this PR.

File: mmv1/third_party/terraform/tests/resource_compute_instance_group_manager_test.go.erb (hunk 1 above)
Comment (ilonalit) [reply mmv1/third_party/terraform/tests/resource_compute_instance_group_manager_test.go.erb:634]: Thank for pointing, updated tests

**Comment by hao-nan-li** [comment]
LGTM in general.

**Review by c2thorn — APPROVED** [review]
looks good on my end.

**Pull request merged**